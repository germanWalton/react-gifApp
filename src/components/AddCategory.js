import React, { useState } from "react";
import PropTypes from "prop-types";

const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState("");

  const handleInputChange = (e) => {
    setInputValue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    if (inputValue.trim().length > 2) {
      setCategories((cats) => [inputValue, ...cats]);
      setInputValue("");
    }
  };

  return (
      <form onSubmit={handleSubmit}>
        <div className="d-flex justify-content-center mt-5">
          <input
            className="form-control w-50 p-3"
            type="text"
            value={inputValue}
            placeholder={"Ingrese un nombre"}
            onChange={handleInputChange}
          ></input>
          <button className="btn btn-lg btn-secondary p-3" type="submit">
            Buscar
        </button>
        <button className="btn btn-lg btn-dark ms-3 p-3" type="reset" onClick={() => { setCategories([""]) } }>
          Borrar busqueda
          </button>
        </div>
        <br />
      </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};

export default AddCategory;
