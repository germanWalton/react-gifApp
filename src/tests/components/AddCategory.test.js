import { shallow } from "enzyme";
import React from "react";
import AddCategory from "../../components/AddCategory";
import '@testing-library/jest-dom'


describe('Pruebas en <AddCategory />', () => {

  const setCategories = jest.fn()
  let wrapper = shallow(<AddCategory setCategories={setCategories} />);
  
  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setCategories={setCategories} />);

  })

  test('debe mostrarse correctamente', () => {
    
    expect(wrapper).toMatchSnapshot();
  });
  

  test('debe cambiar la caja de texto', () => {
    
    const input = wrapper.find('input');
    const value = "Hola mundo";

    input.simulate('change', { target: { value } });
    expect(wrapper.find('p').text().trim()).toBe(value);
    
  });

  test('no debe postear la informaciÃ³n con submit', () => {
    
    wrapper.find('form').simulate('submit',{ preventDefault(){} })
   
    expect(setCategories).not.toHaveBeenCalled()
    
  })

  test('debe llamar el setCategories y limpiar la caja de texto', () => {

    const value = "Hola mundo";
    wrapper.find('input').simulate('change', { target: { value } });
    wrapper.find('form').simulate('submit', { preventDefault() { } });

    expect(setCategories).toHaveBeenCalled();
    expect(setCategories).toHaveBeenCalledTimes(1);
    expect(setCategories).toHaveBeenCalledWith(expect.any(Function));

    expect(wrapper.find('input').prop('value')).toBe('');

    
  })
  
})