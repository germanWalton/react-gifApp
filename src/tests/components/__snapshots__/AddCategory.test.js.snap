// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCategory /> debe mostrarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="d-flex justify-content-center mt-5"
  >
    <p />
    <input
      className="form-control w-50 p-3"
      onChange={[Function]}
      placeholder="Ingrese un nombre"
      type="text"
      value=""
    />
    <button
      className="btn btn-lg btn-secondary p-3"
      type="submit"
    >
      Buscar
    </button>
    <button
      className="btn btn-lg btn-dark ms-3 p-3"
      onClick={[Function]}
      type="reset"
    >
      Borrar busqueda
    </button>
  </div>
  <br />
</form>
`;
